platform/ui-next/src/components/SegmentationTable/SegmentationSegments.tsx

```diff
@@ -38,108 +40,148 @@ export const SegmentationSegments = ({ children = null }: { children?: React.Rea
   }
 
   const segments = Object.values(representation.segments);
+
+  // Find the active segment to scroll to it when it changes
+  const activeSegment = segments.find(segment => {
+    if (!segment) {
+      return false;
+    }
+    const segmentFromSegmentation = segmentation.segments[segment.segmentIndex];
+    return segmentFromSegmentation?.active;
+  });
+
   const isActiveSegmentation = segmentation.segmentationId === activeSegmentationId;
 
   const { ref: scrollableContainerRef, maxHeight } = useDynamicMaxHeight(segments);
 
+  useEffect(() => {
+    if (activeSegment) {
+      const activeSegmentRef = segmentElementMap.current.get(activeSegment.segmentIndex);
+
+      if (!activeSegmentRef) {
+        return;
+      }
+
+      const activeSegmentRect = activeSegmentRef.getBoundingClientRect();
+      const scrollableContainerRect = scrollableContainerRef.current.getBoundingClientRect();
+      if (
+        activeSegmentRect.top > scrollableContainerRect.top &&
+        activeSegmentRect.bottom < scrollableContainerRect.bottom
+      ) {
+        return;
+      }
+
+      activeSegmentRef.scrollIntoView({ behavior: 'smooth', block: 'nearest' });
+    }
+  }, [activeSegment, scrollableContainerRef]);
+
   if (!representation || !segmentation) {
     return null;
   }
 
   return (
-    <ScrollArea
-      className={`bg-bkg-low space-y-px`}
-      showArrows={true}
-    >
-      <div
-        ref={scrollableContainerRef}
-        style={{ maxHeight: maxHeight }}
+    <div ref={scrollableContainerRef}>
```

Comment:
Again indentation by this added `div` which is what we now use to calculate the size using the existing `useDynamicMaxHeight` hack. Using the actual scroll container itself is too unpredictable.